ec2/sc-ec2-linux-jumpcloud.yaml

```diff
@@ -0,0 +1,517 @@
+                [main]
+                stack=${AWS::StackId}
+                region=${AWS::Region}
+                verbose=true
+                interval=5
+              mode: "000400"
+              owner: root
+              group: root
+            /etc/cfn/hooks.d/cfn-auto-reloader.conf:
+              content: !Sub |
+                [cfn-auto-reloader-hook]
+                triggers=post.update
+                path=Resources.Ec2Instance.Metadata.AWS::CloudFormation::Init
+                action=/opt/aws/bin/cfn-init -v --stack ${AWS::StackName} --resource Ec2Instance --configsets SetupCfn,SetupJumpcloud,SetupVolume --region ${AWS::Region}
+              mode: "000400"
+              owner: root
+              group: root
+            /lib/systemd/system/cfn-hup.service:
+              content: !Sub |
+                [Unit]
+                Description=cfn-hup daemon
+
+                [Service]
+                Type=simple
+                ExecStart=/opt/aws/bin/cfn-hup
+                Restart=always
+
+                [Install]
+                WantedBy=multi-user.target
+              mode: "000400"
+              owner: root
+              group: root
+          commands:
+            01_enable_cfn-hup:
+              command: "/bin/systemctl enable cfn-hup.service"
+            02_start_cfn-hup:
+              command: "/bin/systemctl start cfn-hup.service"
+        install_jc:
+          packages:
+            !If
+            - IsUbuntu
+            -
+              apt:
+                awscli: []
+                jq: []
+            -
+              yum:
+                jq: []
+          commands:
+            01_jumpcloud_agent:
+              command: !Join
+                - ''
+                - - "/usr/bin/curl --silent --show-error --header 'x-connect-key: "
+                  - "$(aws ssm get-parameters --names '/infra/JcConnectKey --with-decryption' | jq '.[] | .[0].Value')"
+                  #- '{{resolve:ssm-secure:/infra/JcConnectKey:1}}'
+                  - "' https://kickstart.jumpcloud.com/Kickstart | sudo bash"
+        config_jc:
+          commands:
+            # allow agent time to register with jumpcloud account
+            01_wait_registeration:
+              command: "sleep 10"
+            02_associate_jc_systems:
+              command: !Join
+                - ''
+                - - "JC_SYSTEM_ID=$(sudo cat /opt/jc/jcagent.conf|jq -r '.systemKey'); "
+                  - "JC_USER_ID=$(/usr/bin/curl --silent --show-error -X GET https://console.jumpcloud.com/api/systemusers "
+                  - "-H 'Accept: application/json' -H 'Content-Type: application/json' -H 'x-api-key: "
+                  #- '{{resolve:ssm-secure:/infra/JcServiceApiKey:1}}'
+                  - "$(aws ssm get-parameters --names '/infra/JcServiceApiKey --with-decryption' | jq '.[] | .[0].Value')"
+                  - "' | jq -r '.results | .[] | select(.email==\""
+                  - !Ref OwnerEmail
```

Comment:
I'm wondering if there's any way we can automatically get this from the current user SC user that's doing the provisioning?

Reply:
@zaro0508 Not through CloudFormation, unfortunately. I looked into that when we were setting up the trial in IT Sandbox. However, maybe another awcli call here. aws sts has "get-caller-identity" which "Returns details about the IAM user or role whose credentials are used to call the operation." No idea what that's going to return when SC runs the template, though.

Reply:
@zaro0508 I haven't found a way to do this. I can call `aws sts get-caller-identity`, but if this is done when a role has been assumed, as will be the case within ServiceCatalog, then only information about the assumed role is returned. I haven't found a way to reverse engineer this to get the original user.